### DISCUSSION TITLE
How do I delete a path starting from a given commit hash?

### DISCUSSION THREAD

**Discussion by andersonvom** [newren/git-filter-repo#571] [Q&A]
Given the following commit tree (time flows downwards), is there a way to delete a given file path `fp` from all commits in all branches starting from commit `h6`?

```
              (`git init` here)
                     h1
                     h2 ----\
                     h3     h16
                     h4     h17 (branch1)
                     h5
🏁 starting here ->  h6
                     h7-----\
                     h8     h18
                     h9     h19 (branch2)
                     h10
                     h11
                     h12----\
                     h13    h20
                     h14    h21 (branch3)
                     h15
                     h15 (main)
```

**Comment by newren** [comment]
Something like this:

```
git filter-repo --refs ^${hash_of_h6} branch2 branch3 main --commit-callback "commit.file_changes += [
    FileChange(b'D', b'fp')]"
```

If I misinterpreted "starting from h6" (which I took as all commits after h6 but not including h6 itself), you might need to adjust and make that ${hash_of_h5}; it wasn't clear to me.  Also, if you have more branches than just branch2, branch3, and main, you'd need to list them.  If `fp` isn't your actual filename, and e.g. `src/module/file.py` is, you'd need to update the second argument of the FileChange directive (e.g. `FileChange(b'D', b'src/module/file.py')`)

Does that help?
